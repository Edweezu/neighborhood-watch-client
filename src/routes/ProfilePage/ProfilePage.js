import React from 'react'
import { Link } from 'react-router-dom'

class ProfilePage extends React.Component {

    state = {
        error: null
    }

    render () {
        const { error } = this.state

        return (
            <section className='ProfilePage'> 
                 <h2>
                    Create a Profile
                </h2>
                <form className='LoginForm' onSubmit={this.handleSubmit}>
                    <div role='alert'>
                        {error && <p className='red'>{error}</p>}
                    </div>
                    <div className='LoginForm__signupElement'>
                        <div className='LoginForm__signupLabel'>
                            <label htmlFor='firstName'>
                                First Name    
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                        </div>
                        <div className='LoginForm__signupLabel'>
                            <input id='firstName' name='firstName' type='text'
                            required/>
                        </div>
                    </div>
                    <div className='LoginForm__signupElement'>
                        <div className='LoginForm__signupLabel'>
                            <label htmlFor='lastName'>
                                Last Name    
                            </label>
                        </div>
                        <div className='LoginForm__signupLabel'>
                            <input id='lastName' name='lastName' type='text'
                            />
                        </div>
                    </div>
                    <div className='LoginForm__signupElement'>
                        <div className='LoginForm__signupLabel'>
                            <label htmlFor='lastName'>
                                City   
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                        </div>
                        <div className='LoginForm__signupLabel'>
                            <input id='lastName' name='lastName' type='text'
                            required/>
                        </div>
                    </div>
                    {/* <div className='signin-button'>
                        <button className="btn btn-sign-in" type='submit'>
                            Create an Account
                        </button>
                    </div> */}
                    <Link to='/dashboard'>Log In</Link>
                </form>
            </section>
        )
    }
}

export default ProfilePage