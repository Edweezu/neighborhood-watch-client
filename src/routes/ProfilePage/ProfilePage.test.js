import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import ProfilePage from './ProfilePage'


describe('ProfilePage component', () => {
    const props = {
        match: {
            params: {
                userId: 1
            }
          }
    }

    it('renders the given component', () => {
        const wrapper = shallow(<ProfilePage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<ProfilePage {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})