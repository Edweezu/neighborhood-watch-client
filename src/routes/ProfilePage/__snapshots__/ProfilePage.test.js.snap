// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component renders the component's given props 1`] = `
<section
  className="ProfilePage"
>
  <h2>
    Create a Profile
  </h2>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="firstName"
        >
          First Name
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <input
          id="firstName"
          name="firstName"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="lastName"
        >
          Last Name
        </label>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <input
          id="lastName"
          name="lastName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="city"
        >
          Location
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <select
          className="ProfilePage__addSelect countries"
          id="countryId"
          name="country"
          onChange={[Function]}
          required={true}
        >
          <option
            value=""
          >
            Select Country
          </option>
        </select>
        <select
          className="ProfilePage__addSelect states"
          id="stateId"
          name="state"
          onChange={[Function]}
          required={true}
        >
          <option
            value=""
          >
            Select State
          </option>
        </select>
        <select
          className="ProfilePage__addSelect cities"
          id="cityId"
          name="city"
          onChange={[Function]}
          required={true}
        >
          <option
            value=""
          >
            Select City
          </option>
        </select>
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="signin-button"
    >
      <button
        className="btn btn-sign-in"
        type="submit"
      >
        Create an Account
      </button>
    </div>
  </form>
  <Footer />
</section>
`;

exports[`ProfilePage component renders the given component 1`] = `
<section
  className="ProfilePage"
>
  <h2>
    Create a Profile
  </h2>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="firstName"
        >
          First Name
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <input
          id="firstName"
          name="firstName"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="lastName"
        >
          Last Name
        </label>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <input
          id="lastName"
          name="lastName"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="city"
        >
          Location
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <select
          className="ProfilePage__addSelect countries"
          id="countryId"
          name="country"
          onChange={[Function]}
          required={true}
        >
          <option
            value=""
          >
            Select Country
          </option>
        </select>
        <select
          className="ProfilePage__addSelect states"
          id="stateId"
          name="state"
          onChange={[Function]}
          required={true}
        >
          <option
            value=""
          >
            Select State
          </option>
        </select>
        <select
          className="ProfilePage__addSelect cities"
          id="cityId"
          name="city"
          onChange={[Function]}
          required={true}
        >
          <option
            value=""
          >
            Select City
          </option>
        </select>
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="LoginForm__signupLabel"
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="signin-button"
    >
      <button
        className="btn btn-sign-in"
        type="submit"
      >
        Create an Account
      </button>
    </div>
  </form>
  <Footer />
</section>
`;
